### PR TITLE
Support vault namespaces in connect CA

### PR DESCRIPTION
**Pull request by markan**
Follow on to some missed items from #12655

From an internal ticket "Support standard "Vault namespace in the
path" semantics for Connect Vault CA Provider"

Vault allows the namespace to be specified as a prefix in the path of
a PKI definition, but our usage of the Vault API includes calls that
don't support a namespaced key. In particular the sys.* family of
calls simply appends the key, instead of prefixing the namespace in
front of the path.

Signed-off-by: Mark Anderson <manderson@hashicorp.com>

### CODE REVIEW

File: agent/connect/ca/provider_vault_test.go [hunk 1]
```diff
@@ -852,3 +852,34 @@ func vaultProviderConfig(t *testing.T, addr, token string, rawConf map[string]in
 
 	return cfg
 }
+
+func TestVaultProvider_potentialMountPaths(t *testing.T) {
+
```
Comment (kyhavlov) [inline agent/connect/ca/provider_vault_test.go:857]: I think this is a leftover extra whitespace line

File: agent/connect/ca/provider_vault.go [hunk 2]
```diff
@@ -284,7 +286,8 @@ func (v *VaultProvider) GenerateRoot() (RootResult, error) {
 	rootPEM, err := v.getCA(v.config.RootPKIPath)
 	switch err {
 	case ErrBackendNotMounted:
-		err := v.client.Sys().Mount(v.config.RootPKIPath, &vaultapi.MountInput{
+
```
Comment (kyhavlov) [inline agent/connect/ca/provider_vault.go:289]: Leftover extra whitespace line

**Review by kyhavlov — APPROVED** [review]
Looks good 👍 , left a couple non-blocking minor whitespace nits

**Comment by jkirschner-hashicorp** [comment]
Temporarily marking `do-not-merge` while we continue work on this

**Comment by markan** [comment]
Closing this PR in favor of a new approach in PR #12904

**Pull request closed without merging**